DynamicTablesPkg/Library/Acpi/Common/AcpiFadtLib/FadtGenerator.c

```diff
@@ -551,6 +643,246 @@ BuildFadtTable (
     }
   }
 
+  // Get the SCI interrupt from the Platform Configuration Manager

```

Comment:
I think this code block line 645 - 885 should go in an x64 specific FadtArchUpdate function.